src/transformers/data/datasets/glue.py

```diff
@@ -75,9 +74,11 @@ def __init__(
                 "dev" if evaluate else "train", tokenizer.__class__.__name__, str(args.max_seq_length), args.task_name,
             ),
         )
-        with torch_distributed_zero_first(local_rank):
-            # Make sure only the first process in distributed training processes the dataset,
-            # and the others will use the cache.
+
+        # Make sure only the first process in distributed training processes the dataset,
+        # and the others will use the cache.
+        lock_path = cached_features_file + ".lock"
+        with FileLock(lock_path):
```

Comment:
as you suggested a couple of weeks ago @thomwolf, changed this to use Filelock as a way to support more flavors of distributed than just `torch.distributed` (here, TPU cores via xla/xmp)

Will do this for other datasets in a separate PR

Reply:
As a side benefit the API is cleaner too (no more local_rank)